### PR TITLE
Implementation of HTTPCookie.cookies(withResponseHeaderFields:forURL:)

### PR DESCRIPTION
**Pull request by pushkarnk**
This is for an initial implementaion of HTTPCookie.cookies(withResponseHeaderFields:forURL:).
In addition to the contributed unit tests, I have also tested this code against cookies
returned by some of the twenty most popular websites. 

Kindly review and provide feedback.

### CODE REVIEW

File: Foundation/NSHTTPCookie.swift [hunk 1]
```diff
@@ -327,16 +332,106 @@ public class HTTPCookie : NSObject {
...
+            }
+        }
     
+        return httpCookies
+    } 
+
+    //Bake a cookie
+    private class func createHttpCookie(url: URL, pairs: [String], start: Int, end: Int) -> HTTPCookie? {
+        var properties: [String:Any] = [:]
+        for index in start..<end {
```
Comment (parkera) [inline Foundation/NSHTTPCookie.swift]: Any reason not to just use `for aPair in pairs`?

File: Foundation/NSHTTPCookie.swift [hunk 2]
```diff
@@ -459,3 +554,19 @@ public class HTTPCookie : NSObject {
         return _portList
     }
 }
+
+//utils for cookie parsing
+internal extension String {
```
Comment (parkera) [inline Foundation/NSHTTPCookie.swift]: A style note, but I'd rather just put these functions inline at the call site than extending String in the entire project.

**Comment by parkera** [comment]
This looks great, thanks. I'm going to kick off a test run while I review further.

**Comment by parkera** [comment]
@swift-ci please test

File: Foundation/NSHTTPCookie.swift (hunk 1 above)
Comment (parkera) [inline Foundation/NSHTTPCookie.swift]: Oh I see, we're parsing a subset of the array. I think it might be cleaner to just pass an ArraySlice into this.

**Comment by pushkarnk** [comment]
Thanks, I'll continue working on the improvements you suggested

**Pull request merged**